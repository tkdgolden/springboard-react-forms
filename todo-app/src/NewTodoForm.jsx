import { useState } from 'react'

const NewTodoForm = ({ addTodo }) => {
    const INITIAL_STATE = "";
    const [fData, setFormData] = useState({ task: INITIAL_STATE });

    const handleSubmit = evt => {
        evt.preventDefault();

        if (fData.task) {
            addTodo(fData.task);
            setFormData({ task: INITIAL_STATE });
        }
    }

    const handleChange = evt => {
        const { name, value } = evt.target;

        setFormData(fData => ({
            ...fData,
            [name]: value
        }));
    };

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="task">Task: </label>
            <input
                id="task"
                name="task"
                type="text"
                value={fData.task}
                onChange={handleChange}
            >
            </input>
            <button>Enter</button>
        </form>
    );
};

export default NewTodoForm