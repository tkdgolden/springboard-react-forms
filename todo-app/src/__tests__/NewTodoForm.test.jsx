import { describe, expect, test } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom'
import NewTodoForm from "../NewTodoForm";
import { useState } from "react";

describe("NewTodoForm", () => {
    // smoketest
    test("renders", () => {
        render(<NewTodoForm />);
        const title = screen.getByText("Task", { exact: false });
        expect(title.textContent).toBe("Task: ");
    });

    // snapshot
    test("hasn't changed", () => {
        const { asFragment } = render(<NewTodoForm />);
        expect(asFragment()).toMatchSnapshot();
    });

    test("can input task", function () {
        const { getByLabelText, queryByText } = render(<NewTodoForm />);

        // default
        const taskInput = getByLabelText("Task:");
        expect(taskInput.value).toBe("");

        // fill out the form
        fireEvent.change(taskInput, { target: { value: "gym" } });

        // changed
        expect(taskInput.value).toBe("gym");
    });

});