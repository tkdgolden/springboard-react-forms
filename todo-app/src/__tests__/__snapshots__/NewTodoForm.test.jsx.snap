// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewTodoForm > hasn't changed 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      Task: 
    </label>
    <input
      id="task"
      name="task"
      type="text"
      value=""
    />
    <button>
      Enter
    </button>
  </form>
</DocumentFragment>
`;

exports[`TodoList > hasn't changed 1`] = `
<DocumentFragment>
  <h1>
    To Do List
  </h1>
  <div
    id="1"
  >
    <p>
      Tada
    </p>
    <button>
      X
    </button>
  </div>
  <form>
    <label
      for="task"
    >
      Task: 
    </label>
    <input
      id="task"
      name="task"
      type="text"
      value=""
    />
    <button>
      Enter
    </button>
  </form>
</DocumentFragment>
`;
